airflow/jobs/scheduler_job.py

```diff
@@ -1194,7 +1195,7 @@ def _schedule_dag_run(
         return callback_to_run
 
     @provide_session
-    def _verify_integrity_if_dag_changed(self, dag_run: DagRun, session=None):
+    def _verify_integrity_if_dag_changed(self, dag_run: DagRun, session: Session = NEW_SESSION) -> None:
```

Comment:
I think the `provide_session` should be removed here. It's not used and it's not proper to create a new session inside the scheduler loop